crates/typst/src/model/styles.rs

```diff
@@ -397,25 +411,29 @@ impl<'a> StyleChain<'a> {
     }
 
     /// Iterate over all values for the given property in the chain.
-    pub fn properties<T: FromValue + 'a>(
-        self,
-        func: Element,
-        name: &'a str,
-        inherent: Option<Value>,
-    ) -> impl Iterator<Item = T> + '_ {
-        inherent
-            .into_iter()
-            .chain(
-                self.entries()
-                    .filter_map(Style::property)
-                    .filter(move |property| property.is(func, name))
-                    .map(|property| property.value.clone()),
-            )
-            .map(move |value| {
-                value.cast().unwrap_or_else(|err| {
-                    panic!("{} (for {}.{})", err, func.name(), name)
-                })
-            })
+    pub fn properties<'b, T: Blockable>(
+        &'b self,
+        func: ElementData,
+        id: impl Into<u8>,
+        inherent: Option<&'b T>,
+    ) -> impl Iterator<Item = &'b T> + 'b {
+        let id = id.into();
+        inherent.into_iter().chain(
+            self.entries()
+                .filter_map(Style::property)
+                .filter(move |property| property.is(func, id))
+                .map(|property| &property.value)
+                .map(move |value| {
+                    value.downcast().unwrap_or_else(|| {
+                        panic!(
+                            "for {}.{}: {:?}",
+                            func.name(),
+                            func.field_name(id).unwrap(),
+                            value
+                        )
```

Comment:
this panic message is confusing, starting with the `for`. the old one stated the casting error, but this one would leave a user very confused.

Reply:
Very true, I'll improve upon it.